R/rd-html.R

```diff
@@ -34,21 +34,17 @@ flatten_para <- function(x, ...) {
   needs_split <- !is_block & !empty
   html[needs_split] <- purrr::map(html[needs_split], split_at_linebreaks)
 
-  blocks <- html %>%
-    split(groups) %>%
-    purrr::map(unlist) %>%
-    purrr::map_chr(paste, collapse = "")
+  blocks <- purrr::map_chr(split(html, groups), function(x) {
+    paste(unlist(x), collapse = "")
+  })
 
   # There are three types of blocks:
   # 1. Combined text and inline tags
   # 2. Paragraph breaks (text containing only "\n")
   # 3. Block-level tags
   #
   # Need to wrap 1 in <p>
-  needs_p <- (!(is_nl | is_block)) %>%
-    split(groups) %>%
-    purrr::map_lgl(any)
-
+  needs_p <- purrr::map_lgl(split(!(is_nl | is_block), groups), any)
```

Comment:
unsure about the sort of double negation (but I know it was there before)

Reply:
Yeah, I started trying to fix it but decided it was better left for another PR, and I need to rework this function anyway for #2371